rct229/rulesets/ashrae9012019/ruleset_functions/g311_exceptions/does_zone_meet_G3_1_1c.py

```diff
@@ -0,0 +1,138 @@
+    zones_and_systems dict
+        a dict map zone to an object that carries expected_system_type and system type string
+
+    Returns
+    -------
+    boolean, true or false
+    """
+
+    def get_zone_weekly_eflh(z_id):
+        # function to get weekly zone eflh
+        zone_eflh = get_zone_eflh(rmi, z_id, is_leap_year)
+        return (
+            zone_eflh / NUMBER_OF_WEEKS_IN_LEAP_YEAR
+            if is_leap_year
+            else zone_eflh / NUMBER_OF_WEEKS_IN_YEAR
+        )
+
+    expected_system_type = zones_and_systems[zone_id]["expected_system_type"]
+    system_matched = any(
+        [
+            baseline_system_type_compare(
+                expected_system_type,
+                target_system_type,
+                exact_match=False,
+            )
+            for target_system_type in ELIGIBLE_PRIMARY_SYSTEM_TYPES
+        ]
+    )
+
+    if system_matched:
+        zones_on_same_floor_ids = get_zones_on_same_floor_list(rmi, zone_id)
+        # drop zone_id in the list
+        zones_on_same_floor_ids.remove(zone_id)
+        # keep only matched system type
+        zones_same_floor_same_system_type = list(
+            filter(
+                lambda other_zone_id: zones_and_systems[other_zone_id][
+                    "expected_system_type"
+                ]
+                == expected_system_type,
+                zones_on_same_floor_ids,
+            )
+        )
+        # In here, the function assumes the zones_and_systems keys are
+        # a list of conditioned or semi-conditioned zones only
+        zone_load_and_eflh_list = [
+            (
+                get_zone_peak_internal_load_floor_area_dict(rmi, other_match_zone_id),
+                get_zone_weekly_eflh(other_match_zone_id),
+            )
+            for other_match_zone_id in zones_same_floor_same_system_type
+            if other_match_zone_id in zones_and_systems.keys()
+        ]
+
+        system_total_area = sum(map_(zone_load_and_eflh_list, "0.area"), ZERO.AREA)
+        system_total_load = sum(map_(zone_load_and_eflh_list, "0.peak"), ZERO.POWER)
+        avg_eflh = (
+            np.dot(
+                map_(zone_load_and_eflh_list, lambda zl: zl[1]),
+                map_(zone_load_and_eflh_list, lambda zl: zl[0]["area"].magnitude),
+            )
+            / system_total_area.magnitude
+        )
+
+        avg_internal_load = system_total_load / system_total_area
+
+        zone_internal_loads = get_zone_peak_internal_load_floor_area_dict(rmi, zone_id)
+        zone_eflh = get_zone_weekly_eflh(zone_id)
+
+        meet_g3_1_1c_flag = (
+            abs(
+                zone_internal_loads["peak"] / zone_internal_loads["area"]
+                - avg_internal_load
+            )
+            > LOAD_THRESHOLD
+            or zone_eflh - avg_eflh > EFLH_THRESHOLD
+        )
+
+        if meet_g3_1_1c_flag:
+            meet_g3_1_1c_flag = zone_id not in get_zone_computer_rooms(rmi)
+
+    return meet_g3_1_1c_flag
```

Comment:
When ```system_matched``` is False, if statement in line 86 isn't executed, thus ```meet_g3_1_1c_flag``` variable isn't defined. Could you check this out?